cmd/crictl/exec.go

```diff
@@ -178,37 +194,69 @@ var runtimeExecCommand = &cli.Command{
 			transport: c.String(transportFlag),
 		}
 
-		for _, id := range ids {
-			opts.id = id
+		maxParallel := 1
+		if c.Bool("parallel") {
+			maxParallel = runtime.NumCPU()
+		}
+
+		results := mapParallel(ids, maxParallel, func(id string) error {
+			optsCopy := *&opts
+			optsCopy.id = id
 
 			if outputContainerID {
 				fmt.Println(id + ":")
 			}
-
 			if c.Bool("sync") {
-				exitCode, err := ExecSync(runtimeClient, opts)
+				exitCode, err := ExecSync(runtimeClient, optsCopy)
 				if err != nil {
-					return fmt.Errorf("execing command in container synchronously: %w", err)
+					return fmt.Errorf("execing command in container %s synchronously: %w", id, err)
 				}
 				if exitCode != 0 {
 					return cli.Exit("non-zero exit code", exitCode)
 				}
-				continue
+			} else {
+				ctx, cancel := context.WithCancel(c.Context)
+				defer cancel()
+				err = Exec(ctx, runtimeClient, optsCopy)
+				if err != nil {
+					return fmt.Errorf("execing command in container %s: %w", id, err)
+				}
 			}
 
-			ctx, cancel := context.WithCancel(c.Context)
-			defer cancel()
-			err = Exec(ctx, runtimeClient, opts)
-			if err != nil {
-				return fmt.Errorf("execing command in container: %w", err)
-			}
+			return nil
+		})
 
+		if c.Bool("ignore-error") {
+			return nil
```

Comment:
Let's write a warning like "Ignoring errors: errors.Join"

Reply:
Also, it should be probably somewhere before the main execution loop, to let user know ahead of time.

Reply:
Fixed as suggested :+1:

Reply:
Is it a good practice to tell that there were errors?

Reply:
@SergeyKanzhelev, we do that by default. :smile: 

However, when the users passes `--ignore-errors`, then we... ignore errors (you can still see these if you are running debug log level, though). I also suggested warning user once at the beginning when this flag is used for clarity: https://github.com/kubernetes-sigs/cri-tools/pull/1622#discussion_r1793095129

Would that work here for this feature?

Reply:
Note, the current code is slightly different (updates were made).